src/coreComponents/mesh/generators/PAMELAMeshGenerator.cpp

```diff
@@ -38,51 +38,15 @@ namespace geosx
 using namespace dataRepository;
 
 PAMELAMeshGenerator::PAMELAMeshGenerator( string const & name, Group * const parent ):
-  MeshGeneratorBase( name, parent )
-{
-  registerWrapper( viewKeyStruct::filePathString(), &m_filePath ).
-    setInputFlag( InputFlags::REQUIRED ).
-    setRestartFlags( RestartFlags::NO_WRITE ).
-    setDescription( "path to the mesh file" );
-  registerWrapper( viewKeyStruct::fieldsToImportString(), &m_fieldsToImport ).
-    setInputFlag( InputFlags::OPTIONAL ).
-    setDescription( "Fields to be imported from the external mesh file" );
-  registerWrapper( viewKeyStruct::fieldNamesInGEOSXString(), &m_fieldNamesInGEOSX ).
-    setInputFlag( InputFlags::OPTIONAL ).
-    setDescription( "Name of the fields within GEOSX" );
-  registerWrapper( viewKeyStruct::scaleString(), &m_scale ).
-    setInputFlag( InputFlags::OPTIONAL ).
-    setDefaultValue( 1. ).setDescription( "Scale the coordinates of the vertices" );
-  registerWrapper( viewKeyStruct::reverseZString(), &m_isZReverse ).
-    setInputFlag( InputFlags::OPTIONAL ).
-    setDefaultValue( 0 ).setDescription( "0 : Z coordinate is upward, 1 : Z coordinate is downward" );
-}
-
-void PAMELAMeshGenerator::postProcessInput()
-{
-  GEOSX_THROW_IF_NE_MSG( m_fieldsToImport.size(), m_fieldNamesInGEOSX.size(),
-                         "Mismatch between number of values in " << viewKeyStruct::fieldsToImportString() <<
-                         " and " << viewKeyStruct::fieldNamesInGEOSXString() << " attrabutes",
-                         InputError );
-}
-
-Group * PAMELAMeshGenerator::createChild( string const & childKey, string const & childName )
-{
-  GEOSX_THROW( "Invalid child XML node " << childName << " of type " << childKey, InputError );
-}
+  ExternalMeshGeneratorBase( name, parent )
+{}
 
 namespace
 {
 
-string const & getNameSeparator()
-{
-  static string const separator( "_" );
-  return separator;
-}
-
 string makeRegionLabel( string const & regionName, string const & regionCellType )
 {
-  return regionName + getNameSeparator() + regionCellType;
+  return regionName + "_" + regionCellType;
```

Comment:
Sorry, I missed this comment yesterday. So the rationale for getting rid of `getNameSeparator()` is that it was used essentially for two similar, but unrelated purposes: 
  * in `retrieveSurfaceOrRegionName()`, where `"_"` is used as a token separator to truncate some parts of a PAMELA-specific identifier
  * in `makeRegionLabel()`, where it is used to generate a GEOSX-specific cell block name by joining region name with element type name (in fact, it should be renamed to `makeCellBlockName()`).

This second use in principle is completely independent from the first. Currently, by "coincidence" they use the same token separator. So PAMELA region `PART00002_POLYGON_POLYGON_GROUP_Reservoir_1` first turns into `Reservoir_1`, then into `Reservoir_1_TETRA`, but it could also be `Reservoir_1-TETRA`, or `Reservoir_1:TETRA_block` or `TETRA_Reservoir_1`, or... you get the point. 

In other words, the decision on how to construct a GEOSX cell block name _should_ be independent from whatever naming scheme PAMELA uses. In particular, it doesn't necessarily even have to have a single separator - the construction rule could be arbitrary. In this PR, the PAMELA-related naming logic is fully contained in `retrieveSurfaceOrRegionName()`, and GEOSX-related logic is in `makeRegionLabel()`, and there is no dependency between them. In light of that, it no longer makes sense to have a top-level function or constant with the separator - the decision is local to each function.

I hope this makes sense, but I'm open to suggestions.

Reply:
OK, I understand that they're accidentally the same.
For info, did you test if changing GEOSX's `_` into `-` (_e.g._) is working?
Sometimes some accidental relations are used as part of the contract...

Reply:
It affects user input files, specifically `<CellElementRegion cellBlockNames="{ Reservoir_1_TETRA, ... }"/>`. It can also show up in `<FieldSpecification objectPath="..."/>`. So any string that does not include `" ", "/", ","` should work. But yes, it is part of the contract, so we can't just change it freely.